### PR TITLE
Feature configuration of greenprompting llm registry

### LINKED ISSUE #21: LLM Registry provides an Config File for the User
# Feature: Configuration of Greenprompting LLM Registry

## User Story
As a user of the Greenprompting LLM Registry, I want to be able to configure the registry in a simple way so that it has all the necessary information at startup.

## Description
The LLM Registry requires essential configuration information to function properly upon startup. This information should be stored in a `config.ini` file, which must be well-documented, with each entry explaining its purpose and including example values for user reference.

---

## Requirements
1. **Configuration File:**
   - The configuration file should be named `config.ini`.
   - It must contain all necessary entries for the LLM Registry to function.

2. **Documentation:**
   - Each entry in the `config.ini` file must include inline comments explaining its purpose.
   - Example values must be provided for clarity.

3. **Configuration Entries:**
   - **`db_name`:**
     - Description: Specifies the path and name of the SQLite database file.
     - Example Value: `db_name = /path/to/registry_database.db`
     - Note: SQLite creates a database file directly on the filesystem.
   - **`llm_wrapper_machines`:**
     - Description: Specifies a list of host machines where the LLM Registry can deploy LLM Wrappers.
     - Format: A JSON-style array of dictionaries.
     - Example Value:
       ```json
       [{"ip_address": "127.0.0.1",
         "user": "root",
         "password": "toor"}]
       ```

---

## Acceptance Criteria

  Scenario: Validate existence of config.ini file
    Given the project root directory
    When checking for the "config.ini" file
    Then the file exists
    And it contains all required entries with example values

  Scenario: Ensure inline documentation for configuration entries
    Given a valid "config.ini" file
    When reviewing the file contents
    Then each entry has a clear inline comment explaining its purpose

  Scenario: Verify readability of the configuration file
    Given a properly formatted "config.ini" file
    Then it is easy to read and modify

  Scenario: Parse configuration file on startup
    Given a valid "config.ini" file
    When the LLM Registry starts
    Then it successfully parses the configuration
    And utilizes the values correctly

---

### Prerequisites
- A development environment for running the LLM Registry.
- An example `config.ini` file.

### Test Cases
1. **Test Configuration Parsing:**
   - Create a `config.ini` file with valid entries.
   - Start the LLM Registry and verify that the configuration values are correctly parsed and utilized.

2. **Test Missing File:**
   - Remove the `config.ini` file and start the LLM Registry.
   - Verify that the Registry reports an error indicating the missing configuration file.

3. **Test Invalid Configuration:**
   - Modify the `config.ini` file with incorrect values (e.g., invalid JSON for `llm_wrapper_machines`).
   - Verify that the Registry reports an error with a clear message.

4. **Test Valid Configuration:**
   - Use the following example `config.ini` file:
     ```ini
     # Path and name of the SQLite database file
     db_name = /path/to/registry_database.db

     # List of host machines for deploying LLM Wrappers
     llm_wrapper_machines = [{"ip_address": "127.0.0.1", "user": "root", "password": "toor"}]
     ```
   - Start the LLM Registry and confirm successful startup.

5. **Test Edge Cases:**
   - Provide an empty list for `llm_wrapper_machines` and verify that the Registry handles the scenario gracefully.
   - Use a relative path for `db_name` and ensure the database is created in the correct location.

---

### Key Concepts to Understand:
1. **INI File Format:**
   - Learn how to structure an INI file and use inline comments.
2. **Configuration Parsing:**
   - Use Python's `configparser` module to read and parse the configuration file.
   - Example code snippet for parsing:
     ```python
     import configparser

     config = configparser.ConfigParser()
     config.read('config.ini')

     db_name = config['DEFAULT']['db_name']
     llm_wrapper_machines = config['DEFAULT']['llm_wrapper_machines']
     ```
3. **Error Handling:**
   - Implement error handling for missing or malformed configuration files.

### PR DESCRIPTION
**Pull request by AliceKitchkin**
Closes #21

### CODE REVIEW

File: src/app.py
Comment (gflachs) [inline src/app.py:1]: Why did you create a new file called app.py? There is already an file called main.py within the app-folder, which is the entry-point for this app

File: src/app/utils/configreader.py [hunk 1]
```diff
@@ -31,4 +38,38 @@ def get(self, section: str, option: str) -> str:
...
+
+        Args:
+            value (str): The value to validate.
+
+        Raises:
+            ValueError: If the value is not a valid JSON string.
+        """
+        try:
+            parsed_value = json.loads(value)
+            if isinstance(parsed_value, list) and not parsed_value:
```
Comment (gflachs) [inline src/app/utils/configreader.py:57]: If your methode is caled validate json, please validate json and not a not a specific config! Since we dont need to validate the json, as the json modulee already validating the given json via json.load, maybe rename the methode into validate_llm_wrapper_config

File: src/app.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    db_name = config_reader.get("database", "db_name")
+    llm_wrapper_machines = config_reader.get("llm", "llm_wrapper_machines")
+
+    console_logger.info(f"Database Name: {db_name}")
+    console_logger.info(f"LLM Wrapper Machines: {llm_wrapper_machines}")
+
+    return
+
+if __name__ == "__main__":
+    main()
```
Comment (gflachs) [inline src/app.py]: Never do this!

File: src/app.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    
+def main():
+    console_logger.info("Starting LLM Wrapper Service")
+
+    # Initialize ConfigReader
+    config_reader = ConfigReader.get_instance()
+
+    # Retrieve configuration values
+    db_name = config_reader.get("database", "db_name")
```
Comment (gflachs) [inline src/app.py]: Why are you loading the config in here? We dont need it here

File: src/config.ini
Comment (gflachs) [inline src/config.ini:1]: Please use the config.ini within the project root directory. Also please provide different configs for different environments as e.g. local-dev, testing and production

**Review by gflachs — CHANGES_REQUESTED** [review]

**Review by gflachs — APPROVED** [review]
